Add javascript and python engines to wasm worker

tomasol opened the pull request:
Summary: Allow running javascript and python lambda tasks inside WebAssembly worker. WASM resources are taken from wapm.io repository, Wasmer runtime is installed inside the container.

Test Plan: Please follow readme.md where simple workflow with python and javascript tasks is created and executed.

Commit:
Add wasmer and quickJs worker

Commit:
Add python

Commit:
Fix python TODOs, update readme

Commit:
Update readme

Commit:
Disable lambda task

Commit:
Fix UI for showing script expression in ACE editor

Rename script -> scriptExpression in wasm tasks in order to minimize changes
to the UI. Add python syntax highlighting to the ACE editor with language
detection based on task name.

Commit:
Fix reconnecting issues in conductor-client

Commit:
Remove lambda from UI tasks, replace it with JS and PY lambdas

Commit:
Filter out global taskdefs from left menu as they are handled in right sidemenu

Commit:
Fix flow errors

KenG98 commented:
Really excited for this functionality.

I see you're still working on this (new commits coming in) so I'll give very high level feedback. We should avoid committing the python source. I think the approach here is to compile the python interpreter into WASM (which is good).

Instead of committing the Python source, could we do a `git clone` of the python source in the dockerfile during build? So the build would clone the python source and do any relevant wasm building?

Thanks

KenG98 commented:
Also regarding the 5x-20x slowdown of running python through wasm. I think we might get a major performance win by using a small subset of python. Using vanilla python includes a lot of stuff our users shouldn't be using, like the server library, some concurrency libraries, etc. There is probably some open source project which provides a very bare-bones python interpreter that would run faster in WASM.

For now let's stick to vanilla python as-is, and just keep in mind for when performance becomes an issue that that's an option.

Commit:
Replace argv with $.lambdaValue, allow returning result to maintain backwards compatibility with lambda tasks

Commit:
Replace python lib sources with git clone in Dockerfile

Commit:
Increase wasm security by using unprivileged user, fix python directory permissions issue

Run node and wasmer using unprivileged user.
Python needs writable lib folder - create it for each execution

tomasol commented:
I've removed the `lib` folder from this PR and constructing it in Dockerfile as suggested. Apparently python requires this folder to be writable so I have to copy it to a temp directory for every execution - this adds more latency, currently it takes around **1s** to run a simple python script. I agree we should investigate projects like micropython or rustpython in the long term.

KenG98 commented on symphony/app/fbcnms-projects/workflows-proxy/src/proxy/utils.js:
```diff
@@ -42,7 +42,6 @@ const SYSTEM_TASK_TYPES: Array<string> = [
   'EXCLUSIVE_JOIN',
   'WAIT',
   'DYNAMIC',
-  'LAMBDA',
```
Are the wasm workers ready enough that we can disable lambda right away?

tomasol replied:
It is up to discussion if we want to do it now, but there is no way to fix the security problems associated with lambdas.

KenG98 reviewed (commented):
What are `python.wasm` and `quickjs.wasm` and where are they from? Can they also be generated / downloaded in the `docker build` step? I'd like to avoid committing large binaries if possible.

Commit:
Build quickjs.wasm inside docker

Commit:
Download python.wasm inside docker

tomasol commented:
I am building quickjs inside docker now, however due to an [open issue](https://github.com/wapm-packages/python/issues/1) I am just downloading python.wasm from its git repository.

Commit:
Fix workflow proxy test after lambda task removal

Commit:
Merge remote-tracking branch 'fb/master' into wasm-worker-js-py

Commit:
Merge remote-tracking branch 'fb/master' into wasm-worker-js-py
